Sources/SwiftDocC/DocumentationService/Convert/ConvertService+DataProvider.swift

```diff
@@ -50,11 +63,11 @@ extension ConvertService {
         /// The URL this function generates for a resource is not derived from the resource itself, because it doesn't need to be. The
         /// ``DocumentationWorkspaceDataProvider`` model revolves around retrieving resources by their URL. In our use
         /// case, our resources are not file URLs so we generate a URL for each resource.
-        static private func createURL(isMarkupFile: Bool) -> URL {
+        static private func createURL(pathExtension: String? = nil) -> URL {
             var url = URL(string: "docc-service:/\(UUID().uuidString)")!
             
-            if isMarkupFile {
-                url.appendPathExtension(DocumentationBundleFileTypes.referenceFileExtension)
+            if let pathExtension {
```

Comment:
I believe we still guarantee compatibility with Swift 5.5.

```suggestion
            if let pathExtension = pathExtension {
```